src/app/clusters/user-label-server/user-label-cluster.cpp

```diff
@@ -96,20 +85,22 @@ CHIP_ERROR WriteLabelList(const ConcreteDataAttributePath & path, AttributeValue
 
     if (!path.IsListItemOperation())
     {
-        DeviceLayer::AttributeList<Structs::LabelStruct::Type, DeviceLayer::kMaxUserLabelListLength> labelList;
+        size_t numLabels = 0;
+        std::array<Structs::LabelStruct::Type, DeviceLayer::kMaxUserLabelListLength> labels;
         LabelList::TypeInfo::DecodableType decodablelist;
 
         ReturnErrorOnFailure(decoder.Decode(decodablelist));
-        VerifyOrReturnError(IsValidLabelEntryList(decodablelist), CHIP_IM_GLOBAL_STATUS(ConstraintError));
-
         auto iter = decodablelist.begin();
         while (iter.Next())
         {
-            ReturnErrorOnFailure(labelList.add(iter.GetValue()));
+            auto & label = iter.GetValue();
+            VerifyOrReturnError(IsValidLabelEntry(label), CHIP_IM_GLOBAL_STATUS(ConstraintError));
```

Comment:
User label cluster is a code driven clusters, so it has testing infrastructure for unit tests.

Could we test this? Codecov says 58.33333% line coverage.